### DISCUSSION TITLE
Multi-Language Manifest Support

### DISCUSSION THREAD

**Discussion by cbratschi** [vercel/next.js#63330] [Ideas]
### Goals

1. the currently used manifest only support a single language
2. add support for multiple languages

### Background

Manifests should be available in multiple languages. Otherwise multi-language websites can only use texts and URLs for a single language.

Current implementation:

https://nextjs.org/docs/app/api-reference/file-conventions/metadata/manifest

The W3C has a draft extension:

https://www.w3.org/TR/localizable-manifests/

### Proposal

Support an interim solution until the official specs are supported by all major browser:

1. The best way would be to support https://www.w3.org/TR/localizable-manifests/#example-5 and split the manifest into separate file.
2. Add all manifest-<lange>.webmanifest to HTML head.

**Comment by jan10** [comment]
As of April 24, 2026, Chrome/Edge support manifest localization (*_localized fields) starting in version 148, so you can localize name, short_name, description, icons, and shortcuts directly in the manifest.

https://developer.chrome.com/blog/manifest-localization
